File: Frontend/src/app/foodCalorie.ts
import { Time } from "@angular/common";

export class foodCalorie{
    food_id: number | undefined;
    item_type: string ='';
    date: Date  | undefined;
    time: Time | undefined;
    item_name: string = '';
    quantity: number =0;
    calories: number =0;
    protein: number =0;
    carbohydrates: number =0;
    fat: number =0;
}

File: CalorieTrackerApp/src/main/java/com/ai/calorieTrackerApp/models/foodCalorie.java
package com.ai.calorieTrackerApp.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
//import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name = "foodCalorie")
public class foodCalorie {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long foodId;

    @ManyToOne
    @JoinColumn(name = "userId")
    @JsonIgnore
    private userModel user;
    private  String itemType;
    private LocalDate date;
    private LocalTime time;

    private String itemName;

    private double quantity;

    private double calories;

    private double protein;

    private double carbohydrates;
    private double fat;

}
